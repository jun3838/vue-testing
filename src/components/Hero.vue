<template>
  <div class="hero" :style="{ 'backgroundImage': `url(${background})` }">
    <h1 v-text="text"></h1>
  </div>
</template>

<script>
  export default {
    name: 'Hero',
    props: ['text', 'background']
  };
</script>

<style scoped lang="scss">
  .hero {
    background: no-repeat center center/cover;
    padding: rem-calc(20);
    color: #f4f4f4;
    display: flex;
    flex-flow: column nowrap;
    height: 40vh;
    align-items: center;
    justify-content: center;
  }
</style>
