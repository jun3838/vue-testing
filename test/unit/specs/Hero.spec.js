import {shallow} from 'vue-test-utils';
import {createRenderer} from 'vue-server-renderer';
import Hero from '@/components/Hero';

describe('Hero component', () => {
  it('should render with correct props', () => {
    let $mounted = shallow(Hero, {
      propsData: {
        text: 'Testing',
        background: '//picsum.photos/600/600',
      },
    });
    expect($mounted.find('h1').text()).toEqual('Testing');
    expect($mounted.find('.hero').html()).toContain('//picsum.photos/600/600');
  });

  test('Snapshot', () => {
    const renderer = createRenderer();
    const wrapper = shallow(Hero);

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
