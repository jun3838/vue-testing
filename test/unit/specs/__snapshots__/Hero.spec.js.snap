// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component Snapshot 1`] = `"<div data-server-rendered=\\"true\\" class=\\"hero\\" style=\\"background-image:url(undefined);\\"><h1></h1></div>"`;
